src/components/NewCommentForm.tsx

```diff
@@ -2,102 +1,206 @@
+import {
+  ChangeEvent,
+  Dispatch,
+  FC,
+  FormEvent,
+  SetStateAction,
+  useState,
+} from 'react';
+import cn from 'classnames';
+
+import { Comment } from '../types/Comment';
+import { addComment } from '../utils/api';
+
+const initialFormData = { name: '', email: '', body: '' };
+const initialFromErrors = { name: false, email: false, body: false };
+
+interface Props {
+  postId: number;
+  updateComments: Dispatch<SetStateAction<Comment[]>>;
+}
+
+export const NewCommentForm: FC<Props> = ({ postId, updateComments }) => {
+  const [formData, setFormData] = useState(initialFormData);
+  const [formErrors, setFormErrors] = useState(initialFromErrors);
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const handleChangeField = (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    const { name, value } = event.target;
+
+    setFormData(prevData => ({ ...prevData, [name]: value.trimStart() }));
+  };
+
+  const handleResetForm = () => {
+    setFormData(initialFormData);
+    setFormErrors(initialFromErrors);
+  };
+
+  const handleFormSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const { name, email, body } = formData;
+    const trimmedData = {
+      name: name.trim(),
+      email: email.trim(),
+      body: body.trim(),
+    };
+
+    setFormErrors({
+      name: !trimmedData.name,
+      email: !trimmedData.email,
+      body: !trimmedData.body,
+    });
+
+    if (!trimmedData.name && !trimmedData.email && !trimmedData.body) {
+      return;
+    }
+
+    setIsLoading(true);
+
+    addComment({
+      postId,
+      ...trimmedData,
+    })
+      .then(comment => {
+        updateComments(prevComments => [...prevComments, comment]);
+        setFormData(prevState => ({ ...prevState, body: '' }));
```

Comment:
There is a syntax error in this line. It seems like you intended to spread `prevState`, but the correct syntax is `{ ...prevState, body: '' }`.